src/storage/src/source/mysql/replication.rs

```diff
@@ -110,84 +118,113 @@ pub(crate) fn render<G: Scope<Timestamp = TransactionId>>(
 
             // Get the set of GTIDs currently available in the binlogs (GTIDs that we can safely start replication from)
             let binlog_gtid_set: String = conn.query_first("SELECT GTID_SUBTRACT(@@global.gtid_executed, @@global.gtid_purged)").await?.unwrap();
-            // NOTE: The assumption here for the MVP is that there is only one source-id in the GTID set, so we can just
-            // take the transaction_id from the first one. Fix this once we move to a partitioned timestamp.
-            let binlog_start_gtid = GtidSet::from_str(&binlog_gtid_set)?.first().expect("At least one gtid").to_owned();
-            let binlog_start_transaction_id = TransactionId::new(binlog_start_gtid.earliest_transaction_id());
-
+            // For each GTID UUID in the binlog set, get a partitioned timestamp representing the transaction-id of the first transaction that is available
+            let binlog_start_gtids: BTreeMap<Uuid, _> = match gtid_singletons(GtidSet::from_str(&binlog_gtid_set)?, false) {
```

Comment:
I'm not sure if I'm misinterpreting, but I think it'll be slightly more complicated than just implementing `parse_gtid_set(s: &str)`, since depending on the context we either need to use the lowest or highest transaction-id in the gtid-set string to represent the frontier we care about.

When we parse the `@gtid_executed` value for creating the snapshot frontier, we want the frontier to represent the transactions that will occur after that snapshot point. And for the `resume_upper` the frontier represents all the transactions that might occur in the future.

But when parsing this `binlog_gtid_set` str, we want this antichain to represent the start of transactions that are in the binlog, so that we can compare `PartialOrder::less_equal(&binlog_since, &resume_upper)`. We don't need or care about the future transactions that are beyond what's in the binlog.

So potentially it'd look something like this:
```
enum GtidSetParseOpt {
    // Creates an antichain representing the start transaction-id of each source-id in the set.
    // This will not include range partitions for non-existent source-ids.
    Start,

    // Creates an antichain representing all transactions that are yet to be seen beyond this set.
    // This will include range partitions representing the 'Absent' transactions of source we have not yet seen.
    After,
}

fn parse_gtid_set(gtid_set_str: &str, opt: GtidSetParseOpt) -> Result<Antichain<PartitionedGtid>, Error>
```

Reply:
I think this complication is a sideeffect of requesting `GTID_SUBTRACT(@@global.gtid_executed, @@global.gtid_purged)` from MySQL. It sounds to me that what we want is the pure `@@global.gtid_purged` which we convert into a frontier in the usual way. This is will be the upper frontier beyond which all updates are available on the server. So we'll end

Reply:
Yes you're totally right! That definitely simplifies a lot of things -- thanks!